addon/components/bs-button.js

```diff
@@ -432,66 +370,32 @@ export default class Button extends Component {
    */
   @action
   handleClick(e) {
-    let onClick = this.onClick;
-    let preventConcurrency = this.preventConcurrency;
+    let onClick = this.args.onClick;
+    let preventConcurrency = this.args.preventConcurrency;
 
-    if (onClick === null || onClick === undefined) {
+    if (!onClick) {
       return;
     }
 
     if (!preventConcurrency || !this.isPending) {
-      let promise = (onClick)(this.value);
+      let promise = (onClick)(this.args.value);
       if (promise && typeof promise.then === 'function' && !this.isDestroyed) {
-        this.set('state', 'pending');
+        this.state = 'pending';
         promise.then(() => {
           if (!this.isDestroyed) {
-            this.set('state', 'fulfilled');
+            this.state = 'fulfilled';
           }
         }, () => {
           if (!this.isDestroyed) {
-            this.set('state', 'rejected');
+            this.state = 'rejected';
           }
         }
         );
       }
     }
 
-    if (!this.bubble) {
+    if (!this.args.bubble) {
       e.stopPropagation();
     }
   }
-
-  init() {
-    super.init(...arguments);
-
-    // deprecate arguments used for attribute bindings only
-    runInDebug(() => {
```

Comment:
The warning that the arguments have been removed shouldn't be removed, should it?

Reply:
Well, not sure when this could be released. My initial assumption was as part of v5, so in that case maybe we can remove those warnings as they have been part of the v4.x cycle. But not sure, again this PR is not really ready to be merged, more an experiment to get some learnings...

Reply:
I added them again now!